test/errors-spec.js

```diff
@@ -6,76 +6,76 @@ describe("Parser:", function() {
 
     describe("#handleInfixNodes", function() {
         it("rejects repeated infix operators", function() {
-            expect("1\\over 2\\over 3").toFailWithParseError(
+            expect`1\over 2\over 3`.toFailWithParseError(
```

Comment:
This looks a little strange to me b/c I'm not used to template strings returning things that aren't strings.  Maybe we could introduce a `tex` template literal function.  It would look like:
```
expect(tex`1\over 2\over 3`).toFailWithParseError(...
```
The `tex` function could be exported for use by other people.  Also, it would be easier to understand what's going on.  I'm still not sure how `expect` is acting as a template function without us modifying/replacing it with our own implementation.

Reply:
@kevinbarabash I agree that it can be unfamiliar since it's new syntax. However, it's new way to call the function with arguments: an array of strings with `raw` property, an array of raw strings, and expressions in `${}`. As `expect` accepts any object, it just passes the first argument through and we can handle them in the matchers. Maybe we can document them in the `CONTRIBUTING.md`?

Reply:
If `expect` passes through only one argument, does that mean this won't work with `${interpolated expressions}`?

I don't think the new code *looks* much more confusing than the old code -- the weirdest part is that `expect` just passes on its argument, and the action is in the chained methods afterward -- but that's a general Jest thing.  However, until I read the [template literal description](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) I didn't really understand what the code was doing.

What about `expectTex`1 \over 2 \over 3` which is equivalent to `expect(tex`1 \over 2 \over 3`)`, i.e., the string gets processed before calling `expect`, instead of requiring all matches to process it?  I think the weirdest part is overloading `expect` in this unintended way (unless Jest suggests doing this?).

Reply:
It's not related, but Jest suggests a way of [creating a table using tagged template literal](https://jestjs.io/docs/en/api#2-describeeach-table-name-fn):

```js
describe.each`
  a    | b    | expected
  ${1} | ${1} | ${2}
  ${1} | ${2} | ${3}
  ${2} | ${1} | ${3}
`('$a + $b', ({a, b, expected}) => {
  test(`returns ${expected}`, () => {
    expect(a + b).toBe(expected);
  });

  test(`returned value not be greater than ${expected}`, () => {
    expect(a + b).not.toBeGreaterThan(expected);
  });

  test(`returned value not be less than ${expected}`, () => {
    expect(a + b).not.toBeLessThan(expected);
  });
});
```

Reply:
Interesting.  I was unaware of jest's use of tagged template literals.  `expectTex` seems like a good compromise.

Reply:
@ylemkimon apologies for not raising my concerns about `expect` as a tagged template earlier.

Reply:
@kevinbarabash @edemaine 

> @ylemkimon apologies for not raising my concerns about `expect` as a tagged template earlier.

I don't mind.

> What about ```expectTex`1 \over 2 \over 3` ``` which is equivalent to ```expect(tex`1 \over 2 \over 3`)```, i.e., the string gets processed before calling `expect`, instead of requiring all matches to process it?  I think the weirdest part is overloading `expect` in this unintended way (unless Jest suggests doing this?).

> Interesting.  I was unaware of jest's use of tagged template literals.  `expectTex` seems like a good compromise.

I still disagree. The purpose of using tagged template literal is to avoid using escape backslashes and simplify test cases. I think using `expectTeX` defeats this purpose as there is no difference between processing before passing to `expect` and processing in the matchers.

Reply:
@ylemkimon There would be a difference if we needed to use Jest's matchers.  But I can't actually think of any situation where we'd want to use those on a TeX string.